common/lib/xmodule/xmodule/video_module/video_module.py

```diff
@@ -232,7 +232,8 @@ def get_html(self):
                     if url:
                         if url not in sources:
                             sources.append(url)
-                        if self.download_video:
+                        # don't include hls urls for download
+                        if self.download_video and not url.endswith('.m3u8'):
```

Comment:
I think we can apply same check at line 269 to not include hls url to download_video_link.